### ISSUE TITLE
Validate backend data request and improve responsiveness

### ISSUE THREAD

**Issue by stackoversnow** [OpenCovenant/quill-fe#6]
The UI seems a bit too reliant on backend data. In such cases the UI seems to still refresh somehow. That's why it is necessary to validate backend data before initiating a UI-Refresh (DOM re-rendering), which not only slows down but also produces plenty of un-anticipated 'undefined' console errors. 

Suggested idea depicted below:
- Validation of data can be done in FE 
- Invalid Data = undefined, empty string, dead server, pending request,.etc
- Give user feedback if server is "currently unavailable"
![diagram111](https://user-images.githubusercontent.com/99611775/156860516-fa1456d2-3e7f-4b73-8418-7f4db0675da1.png)

**Comment by AndiBraimllari** [comment]
Agreed that the FE is very reliant on the data fetched from the calls. We can certainly make it slightly more independent and can do a better job communicating when things go astray. I'd say we keep this issue open and keep an eye on it during development, sounds good?
